src/main/java/com/couchbase/lite/replicator/BlobRequestBody.java

```diff
@@ -42,6 +44,36 @@ public MediaType contentType() {
                 return contentType;
             }
 
+
+            /**
+             * OkHttp use chunked transfer if content-length is unknown. And CouchDB 1.6.1 can not
+             * handle chunked transer with multipart doc, then CouchDB disconnect .
+             * To send non-chunked transfer with multipart document by OkHttp, the client needs
+             * to specify content-length.
+             * Sync Gateway can accept chunked transfer
+             */
+            @Override
+            public long contentLength() throws IOException {
+                // Sync Gateway
+                if (syncGateway) {
+                    return super.contentLength();
+                }
```

Comment:
I am not sure if chunked transfer is preferred method for Sync Gateway. But Sync Gateway can handle chunked transfer, and we tested it for a while. So I consider it is not good time to switch.

I also talked with @adamcfraser. Sync Gateway does not care if it is chunked transfer or not.